README.md

```diff
@@ -1,5 +1,115 @@
-# ts-npm-package-boilerplate
+# cleanup-registry
+a package for calling async functions to be resolved in a configured loop
 
-After cloning this template, please do the following:
-1. insert secrets to repo secrets for the github actions.
-2. replace every string "ts-npm-package-boilerplate" with your package name.
+# Install
+```
+npm i cleanup-registry
+```
+
+# Usage
+1. create your registry with wished options
+```ts
+const registry = new CleanupRegistry({
+      overallTimeout: 3000,
+      preCleanup: async (): Promise<void> => {
+        logger.debug({ message: 'this is a pre cleanup function'})
+        return new Promise((resolve, _) => resolve());
+      },
+    });
+```
+
+2. listen to your wanted events
+```ts
+registry.on('started', () => logger.info({ msg: 'cleanup started'}));
+registry.on('itemCompleted', (id) => logger.info({ msg: `item ${id.toString()} completed` }));
+registry.on('itemFailed', (id, error, message) => logger.error({ err: error, msg: message, id }));
+registry.on('finished', (status) =>
+    status === 'success' ?
+    logger.info(`finished successfully`) :
+    logger.error({ msg: `finished with bad status`, status }));
+```
+
+3. register your async functions
+```ts
+registry.register({
+    func: myAsyncFunc,
+    id: 'funcId',
+    timeout: 100,
+    timeoutAfterFailure: 300
+});
+```
+
+4. trigger the registered functions
+```ts
+await registry.trigger();
+```
+
+# Configuration
+## RegistryOptions
+- `overallTimeout`: the duration in ms for all the registered functions to complete. when timeout duration is passed new function completion attempts will cease and a `finished` event with `timeout` status will be emitted.
+defaults to 10000.
+- `preCleanup`: an async function to be called upon triggering the registry before any of the registered functions will be called.
+- `postCleanup`: an async function to be called after all registerd functions have completed.
```

Comment:
1. add hook to the name as it part of what it does (yes, edit the code.
2. no need for the upon `triggering`

Reply:
done.